src/ucp/rndv/proto_rndv.inl

```diff
@@ -243,31 +243,28 @@ ucp_proto_rndv_bulk_max_payload(ucp_request_t *req,
     size_t total_offset = ucp_proto_rndv_request_total_offset(req);
     size_t total_length = ucp_proto_rndv_request_total_length(req);
     size_t max_frag_sum = rpriv->mpriv.max_frag_sum;
-    size_t lane_offset, max_payload, scaled_length;
-    size_t min_rndv_chunk_size;
+    size_t lane_offset, max_payload, scaled_length, length;
 
     if (ucs_likely(total_length < max_frag_sum)) {
         /**
          * Each lane sends less than its maximal fragment size but more than the
          * minimal chunk size
          */
-        min_rndv_chunk_size =
-                req->send.ep->worker->context->config.ext.min_rndv_chunk_size;
         scaled_length = ucp_proto_multi_scaled_length(lpriv->weight_sum,
                                                       total_length);
+        length        = ucs_max(scaled_length, lpriv->min_end_offset);
```

Comment:
rename length to "end_offset"
could also rename "scaled_length" to "weight_offset" or similar..

Reply:
fixed